FireboltNETSDK/Client/FireboltConnectionSettings.cs

```diff
@@ -97,5 +97,19 @@ internal FireboltConnectionSettings(FireboltConnectionStringBuilder builder)
             }
             return (endpoint ?? Constant.DEFAULT_ENDPOINT, env ?? Constant.DEFAULT_ENV);
         }
+
+        private string GetOneValue(string firstName, string secondName, string? firstValue, string? secondValue)
+        {
+            string[] notNullValues = new string?[] { firstValue, secondValue }.Where(s => !string.IsNullOrEmpty(s)).Select(s => s!).ToArray();
+            int notNullCount = notNullValues.Count();
+            switch (notNullCount)
+            {
+                case 0: throw new FireboltException($"Either {firstName} or {secondName} parameter is missing in the connection string");
+                case 1: return notNullValues[0];
+                case 2: throw new FireboltException($"Ambiguous values of {firstName} and {secondName}. Use only one of them");
+                // this cannot happen. Added to satisfy compiler and to prevent future bugs.
+                default: throw new InvalidOperationException($"{notNullCount} values for {firstName} or {secondName}. Only one value is legal");
+            }
+        }
```

Comment:
I would just split this function into a `GetNotNullValue` and `VerifyInputs`. This way not only the names actually describe what it does but it's also easier to understand what's going on. More visibility on what's happening in the code is better IMO.

Reply:
I have reviewed this function again. I do not know how to split it and avoid code duplication.

Reply:
This is IMO massively over-engineered function in order to save possibly two lines of duplicate code. It does two things, as I mentioned, "coalesce" firstValue or secondValue (as the name suggests but in an obscure way of counting nulls), it also throws errors which will be confusing to see `GetOneValue` complaining about inputs.

Have one function that verifies the connection string, you can then add as many rules there as you want. This would also be extensible, if we were to add some more constraints or verify that some of the inputs are aphanumeric, etc. So something like `private boolean isConnectionValid(FireboltConnectionStringBuilder builder) throws FireboltExcepion`
For the first non-null value, no need to reinvent the wheel - Apache Commons which we already have as a dependency has `firstNonNull` https://commons.apache.org/proper/commons-lang/javadocs/api-3.1/org/apache/commons/lang3/ObjectUtils.html#firstNonNull%28T...%29

Reply:
You're right, not sure why I looked at Java. Regardless, it looks ugly to me too, but not because of GetNotNullValue, I like having simple functions that do one thing. The switch there and counting nulls for two values is unnecessary and just contributes to obscurity. Have a look at the suggestion I've added in a separate comment - much simpler, some duplication but it's clear what's happening from the first glance.